agent/runner/runner_test.go

```diff
@@ -121,28 +117,127 @@ func TestDefaultCapacityLimit(t *testing.T) {
 	t.Parallel()
 
 	// Use default capacity
-	cr := New(0)
+	cr := New(0, 0)
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	go cr.Run(ctx)
 
-	totalJobs := 2 * defaultCapacity
+	totalJobs := 2 * defaultTotalCapacity
 	for i := 0; i < totalJobs; i++ {
 		require.NoError(t, cr.StartJob(testJob{id: fmt.Sprintf("test-%d", i), timeout: time.Second}))
 	}
 
-	// Let first jobs start
+	// Let jobs to start
 	time.Sleep(500 * time.Millisecond)
 
+	var running int
 	for i := 0; i < totalJobs; i++ {
 		// Check that running jobs amount is not exceeded default capacity.
-		assert.Equal(t, i < defaultCapacity, cr.IsRunning(fmt.Sprintf("test-%d", i)))
+		if cr.IsRunning(fmt.Sprintf("test-%d", i)) {
+			running++
+		}
 	}
+
+	assert.Equal(t, defaultTotalCapacity, running)
+}
+
+func TestPerDBInstanceLimit(t *testing.T) {
+	t.Parallel()
+
+	cr := New(10, 1)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go cr.Run(ctx)
+
+	j1db1 := testJob{id: "test-1", timeout: time.Second, dsn: "postgresql://db1"}
+	j2db1 := testJob{id: "test-2", timeout: time.Second, dsn: "postgresql://db1"}
+	j3db1 := testJob{id: "test-3", timeout: time.Second, dsn: "postgresql://db1"}
+	j1db2 := testJob{id: "test-4", timeout: time.Second, dsn: "postgresql://db2"}
+	j2db2 := testJob{id: "test-5", timeout: time.Second, dsn: "postgresql://db2"}
+	j3db2 := testJob{id: "test-6", timeout: time.Second, dsn: "postgresql://db2"}
```

Comment:
I think in these tests it's better to put db in front of job number to make it easier to see that it's another db job. I would say in this context db is more important then job number.
```suggestion
	db1j1 := testJob{id: "test-1", timeout: time.Second, dsn: "postgresql://db1"}
	db1j2 := testJob{id: "test-2", timeout: time.Second, dsn: "postgresql://db1"}
	db1j3 := testJob{id: "test-3", timeout: time.Second, dsn: "postgresql://db1"}
	db2j1 := testJob{id: "test-4", timeout: time.Second, dsn: "postgresql://db2"}
	db2j2 := testJob{id: "test-5", timeout: time.Second, dsn: "postgresql://db2"}
	db2j3 := testJob{id: "test-6", timeout: time.Second, dsn: "postgresql://db2"}
```